### PR TITLE
Exams by Strelia Illia IM-23

### PR DESCRIPTION
**Pull request by Moriartymath**

### CODE REVIEW

File: Tasks/count.js [hunk 1]
```diff
@@ -1,17 +1,12 @@
-// Sum all number values in dict
+'use strict'
 
-count = (obj) => {
-  sum = 0;
-  {
+ const adder = (obj) => {
+  let sum = 0;
+  const arr_of_values = Object.values(obj);
```
Comment (illiakroshka) [inline Tasks/count.js:5]:
```suggestion
  const arrValues = Object.values(obj);
```
You should change the name of variable, the snake case is't good

File: Tasks/intersection.js [hunk 2]
```diff
@@ -1,25 +1,13 @@
...
-      delete object_1[attribute_name];
-      {
-      }
-    }
-    {
-    }
-  }
-  {
+'use strict'
+const intersection =(object_1, object_2)=> {
```
Comment (nevrodegenerate) [inline Tasks/intersection.js:2]: Constant name should be in the uppercase

File: Tasks/count.js [hunk 3]
```diff
@@ -1,17 +1,12 @@
-// Sum all number values in dict
+'use strict'
```
Comment (tshemsedinov) [inline Tasks/count.js:1]:
```suggestion
'use strict';
```

File: Tasks/count.js [hunk 4]
```diff
@@ -1,17 +1,12 @@
...
 
-count = (obj) => {
-  sum = 0;
-  {
+ const adder = (obj) => {
+  let sum = 0;
+  const arrValues = Object.values(obj);
+  for (const element of arr_of_values) {
```
Comment (tshemsedinov) [inline Tasks/count.js:6]: This should not work, `arr_of_values` is not defined
```suggestion
  for (const element of arr_of_values) {
```

File: Tasks/intersection.js [hunk 5]
```diff
@@ -1,25 +1,13 @@
...
-      delete object_1[attribute_name];
-      {
-      }
-    }
-    {
-    }
-  }
-  {
+'use strict'
+const INTERSECTION =(object_1, object_2)=> {
```
Comment (tshemsedinov) [inline Tasks/intersection.js:2]: Use `camelCase`

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
Use eslint and prettier before commit, run tests to be sure it works

File: Tasks/count.js (hunk 4 above)
Comment (Moriartymath) [reply Tasks/count.js:6]: 'use strict';

 const adder = (obj) => {
  let sum = 0;
  const arr_of_values = Object.values(obj);
  for (const element of arr_of_values) {
    if (typeof element === 'number') sum +=   element;
  }
  return sum;
};

require('../Tests/count.js')(adder);

File: Tasks/intersection.js (hunk 5 above)
Comment (Moriartymath) [reply Tasks/intersection.js:2]: 'use strict';

 const adder = (obj) => {
  let sum = 0;
  const arr_of_values = Object.values(obj);
  for (const element of arr_of_values) {
    if (typeof element === 'number') sum +=   element;
  }
  return sum;
};

require('../Tests/count.js')(adder);

Comment (Moriartymath) [reply Tasks/intersection.js:2]: 'use strict';
const intersection =(object_1, object_2)=> {
  const arr_keys = Object.keys(object_1);
  for (const key of arr_keys) {
    object_1[key]===object_2[key]
    ?object_2[key]=object_1[key]
        :delete object_1[key]
  }
  return object_1;
};

require('../Tests/intersection.js')(intersection);

File: Tasks/count.js (hunk 3 above)
Comment (Moriartymath) [reply Tasks/count.js:1]: use strict';

const adder = (obj) => {
let sum = 0;
const arr_of_values = Object.values(obj);
for (const element of arr_of_values) {
if (typeof element === 'number') sum += element;
}
return sum;
};

require('../Tests/count.js')(adder)

File: Tasks/count.js (hunk 4 above)
Comment (Moriartymath) [reply Tasks/count.js:6]: correct variant

File: Tasks/intersection.js (hunk 5 above)
Comment (Moriartymath) [reply Tasks/intersection.js:2]: correct variant

**Pull request closed without merging**